From 2783417c99be6610bdcbfc72b3d80800de26384a Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 17:11:35 -0800
Subject: [PATCH 3000/3005] MacPorts Only: Helper script for generating patches

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 534fe4429053a7bec4fff1ef3de779871602dc93)
---
 create_macports_patches.sh | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100755 create_macports_patches.sh

diff --git llvm_release_35/projects/libcxx/create_macports_patches.sh macports_release_35/projects/libcxx/create_macports_patches.sh
new file mode 100755
index 0000000..938e468
--- /dev/null
+++ macports_release_35/projects/libcxx/create_macports_patches.sh
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+parse_git_branch()
+{
+    ref=$(git symbolic-ref HEAD 2> /dev/null) || return;
+    echo ${ref#refs/heads/}
+}
+
+branch=$(parse_git_branch)
+
+git format-patch origin/${branch} --src-prefix=llvm_${branch}/projects/libcxx/ --dst-prefix=macports_${branch}/projects/libcxx/ --start-number=3000
-- 
2.2.2

